scripts/init.groovy.d/aa_configure_credentials.groovy

```diff
@@ -0,0 +1,370 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: CREDENTIALS ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+
```

Comment:
Is the logic above this comment before the import statements in order to prevent loading unnecessary imports in the case that `autoConfigComplete` already exists?

In other words, was it intentional to have the imports not at the very top to save resources in the case of a no-op?

Reply:
Thats it! yes.